components/SensorComponent.js

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from "react";
+import { Accelerometer } from "expo-sensors";
+import styled from "styled-components/native";
+
+// ==========================
+// = Functions
+const isShaking = (data) => {
+  // x,y,z CAN be negative, force is directional
+  // We take the absolute value and add them together
+  // This gives us the total combined force on the device
+  const totalForce = Math.abs(data.x) + Math.abs(data.y) + Math.abs(data.z);
+
+  // If this force exceeds some threshold, return true, otherwise false
+  // Increase this threshold if you need your user to shake harder
+  return totalForce > 1.78;
```

Comment:
I'm not sure if my phone's accelerometer is very sensitive (on iPhone), but it changes jokes as soon as I just move my phone a little bit. So when I shake it I end up calling 3-4-5 jokes at a time. Maybe increase the threshold above 1.78!